### ISSUE TITLE
Finetuning on custom dataset stops after first epoch

### ISSUE THREAD

**Issue by JanFreise** [asahi417/tner#17]
Hi, 

firstly thank you very much for releasing TNER 🥇 !

I am pretty new to transformers and NER. 
Hopefully you can tell me what i am doing wrong since this can't be an issue with TNER.

I installed TNER and am trying to train completely new NER-Labels/Classifications on a "common" german BERT model available on huggingface.
However after starting the Training is done for one epoch only:

2022-06-17 13:38:54 INFO     target dataset: ['./data']
2022-06-17 13:38:54 INFO     data_name: ./data
2022-06-17 13:38:54 INFO     formatting custom dataset from ./data
2022-06-17 13:38:54 INFO     found following files: {'train': 'train.txt', 'valid': 'valid.txt'}
2022-06-17 13:38:54 INFO     note that files should be named as either `valid.txt`, `test.txt`, or `train.txt` 
2022-06-17 13:38:55 INFO     dataset ./data/train.txt: 60026 entries
2022-06-17 13:38:56 INFO     dataset ./data/valid.txt: 26059 entries
[...]
2022-06-17 14:16:51 INFO     running on 1 GPUs
2022-06-17 14:18:46 INFO     *** start training from step 0, epoch 0 ***
2022-06-17 14:18:47 INFO     [epoch 0] * (training step 0) loss: 2.058, lr: 0.00000200
2022-06-17 13:41:21 INFO     reached maximum step
2022-06-17 13:41:21 INFO     [training completed, 69.66511082649231 sec in total]
2022-06-17 13:41:23 INFO     ckpt saved at ./ckpt_additional_training

This can't be correct, can it? 
I assumed number of epochs is set within the parameter total_step?

Arguments are set like this:

from tner import TrainTransformersNER
trainer = TrainTransformersNER(
        checkpoint_dir='./ckpt_additional_training',  
        dataset='./data',
        transformers_model='dbmdz/bert-base-german-cased',
        random_seed=1234,
        lr=2e-5,
        total_step=100,
        warmup_step=10,
        batch_size=16,
        max_seq_length=256
        )

2022-06-17 14:16:22 INFO     *** initialize network ***
2022-06-17 14:16:22 INFO     create new checkpoint
2022-06-17 14:16:22 INFO     removed incomplete checkpoint ./ckpt_additional_training
2022-06-17 14:16:22 INFO     checkpoint: ./ckpt_additional_training
2022-06-17 14:16:22 INFO      - [arg] dataset: ./data
2022-06-17 14:16:22 INFO      - [arg] transformers_model: dbmdz/bert-base-german-cased
2022-06-17 14:16:22 INFO      - [arg] random_seed: 1234
2022-06-17 14:16:22 INFO      - [arg] lr: 2e-05
2022-06-17 14:16:22 INFO      - [arg] total_step: 100
2022-06-17 14:16:22 INFO      - [arg] warmup_step: 10
2022-06-17 14:16:22 INFO      - [arg] weight_decay: 1e-07
2022-06-17 14:16:22 INFO      - [arg] batch_size: 16
2022-06-17 14:16:22 INFO      - [arg] max_seq_length: 256
2022-06-17 14:16:22 INFO      - [arg] fp16: False
2022-06-17 14:16:22 INFO      - [arg] max_grad_norm: 1.0
2022-06-17 14:16:22 INFO      - [arg] lower_case: False

Also it would be nice to have a parameter for work-in-progess models that are set as "private models" on huggingface which would require to pass use_auth_token = True.

Thanks,
Jan

**Comment by asahi417** [comment]
Hi Jan,

Thank you for playing around TNER, and the option to have a parameter to enable private models with auth token sounds reasonable. I'm working on next version, which is still in progress but I will surely add it in next version.

Regarding to your original question, TNER model training includes a parameter `total_step`, which decides how many step you want to train your model. According to the log you've shared, you set the total_step as 100, meaning model training will be finished in 100 steps. Since your model has 60,026 entries (which is quite massive compared against public NER datasets!) with batch size 16, one epoch (the number of step to run over all the entries once) should be 60,026/16 = 3752, so to train the model `n` epoch, you should set the total_step as `n * 3752`.  If you have larger GPU, you might consider to increase the batch size to decrease the total step. With batch size 128, one epoch reduces to 60,026/128 = 469.

Hope this helps your experiment, and let me know if you have further question or issue.

Thanks

**Comment by JanFreise** [comment]
thanks! now it's working and i learned some basics ^^
also thanks for adding the extra option for huggingface model repo login.

**Comment by asahi417** [comment]
use_auth_token is now available!
